src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts

```diff
@@ -67,15 +68,23 @@ export class CollectionComponent implements OnInit, OnDestroy {
       collectionType: _.get(this.collectionComponentConfig, 'collectionType'),
       collectionStatus: _.get(this.collectionComponentConfig, 'status')
     }, this.sharedContext);
+    this.filters = this.getImplecitFilters();
     this.searchCollection();
     const getCurrentRoleId = _.find(this.programContext.config.roles, {'name': this.sessionContext.currentRole});
     this.sessionContext.currentRoleId = (getCurrentRoleId) ? getCurrentRoleId.id : null;
     this.role.currentRole = this.sessionContext.currentRole;
     this.classes = _.find(this.collectionComponentConfig.config.filters.explicit, {'code': 'gradeLevel'}).range;
-    this.subjects = _.find(this.collectionComponentConfig.config.filters.explicit, {'code': 'subject'}).range;
+    this.mediums = _.find(this.collectionComponentConfig.config.filters.implicit, {'code': 'medium'}).defaultValue;
     this.board = _.find(this.collectionComponentConfig.config.filters.implicit, {'code': 'board'}).defaultValue;
+    (_.size(this.mediums) > 1) ? this.isMediumClickable = true : this.isMediumClickable = false;
   }
 
+  getImplecitFilters(): string[] {
+    const sharedcontext = this.collectionComponentInput.programContext.config.sharedContext,
+    implicitFilter = this.collectionComponentConfig.config.filters.implicit,
+    avilableFileters = this.filterByCollection(implicitFilter, 'code', sharedcontext);
```

Comment:
'available' and 'getImplecitFilters' spelling

Reply:
Changed.